frontend/javascripts/dashboard/dataset/dataset_settings_data_tab.tsx

```diff
@@ -47,12 +51,25 @@ export const syncDataSourceFields = (
   if (syncTargetTabKey === "advanced") {
     // Copy from simple to advanced: update json
     const dataSourceFromSimpleTab = form.getFieldValue("dataSource");
+    if (syncDatasetName && dataSourceFromSimpleTab) {
```

Comment:
The `&& dataSourceFromSimpleTab` was added as `syncDataSourceFields` is called before the first exploration request is sent to the backend. In this case `dataSourceFromSimpleTab==undefined` and thus I guarded this assignment below as else the code would silently crash.